adblock

```diff
@@ -10,12 +10,17 @@ ORIGINAL='/etc/hosts.without-adblock'
 
 # source: http://support.apple.com/kb/HT5343
 clear_dns_cache() {
-	if sw_vers -productVersion | grep -q 10.6
-	then sudo dscacheutil -flushcache
-	else sudo killall -HUP mDNSResponder
-	fi
+	if command -v sw_vers >/dev/null 2>&1
+	then
+		if sw_vers -productVersion | grep -q '10\.6'
+		then sudo dscacheutil -flushcache
+		elif sw_vers -productVersion | grep -q '10\.10[\.123]*'
```

Comment:
10.10.4 seems to be using `mDNSResponder`, as before (at least I can `pgrep` it while I can't `discoveryd`), yet your regexp matches `10.10.4` due to the trailing `*`. Are you sure you meant the `*`?